cmd/juju/application/deployer/bundle_test.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package deployer
+
+import (
+	"github.com/juju/charm/v10"
+	"github.com/juju/juju/core/constraints"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+type bundleSuite struct {
+}
+
+var _ = gc.Suite(&bundleSuite{})
+
+func (s *bundleSuite) TestCheckExplicitBase(c *gc.C) {
+	explicitSeriesError := "series must be explicitly provided when image-id constraint is used"
+
+	testCases := []struct {
+		title         string
+		deployBundle  deployBundle
+		bundleData    *charm.BundleData
+		expectedError string
+	}{
+		// NO IMAGE-ID
```

Comment:
I think all these comments are redundant if the test titles are indicative. I'd just remove them.

Reply:
ok, done